guardrails/validator_base.py

```diff
@@ -175,6 +176,28 @@ class Refrain:
     pass
 
 
+# functions to get chunks
+def split_word(chunk: str):
+    return list(map(lambda x: x + " ", chunk.split(" ")))[:-1]
+
+
+def split_sentence(chunk: str):
+    # using the sentence tokenizer is expensive
+    # we check for a . to avoid wastefully calling the tokenizer
+    if "." not in chunk:
+        return []
+    sentences = nltk.sent_tokenize(chunk)
+    if len(sentences) == 0:
+        return []
+    # return the sentence
+    # then the remaining chunks that aren't finished accumulating
+    return [sentences[0], "".join(sentences[1:])]
+
+
+def split_paragraph(chunk: str):
+    return list(map(lambda x: x + "\n", chunk.split("\n")))[:-1]
```

Comment:
These will need to be class methods.

Reason: Since we're assigning them to the class property `chunking_function`, when this is called via `self.chunking_function` the first parameter that is passed to the function is `self`.  Technically we could add a first parameter that we don't use to capture self and allow the other params to flow through, but it's cleaner to just make all of these class methods instead.

Added bonus: Validator authors can overwrite these with their own splitting logic if they need to.

Reply:
It seems like there's going to be 2 levels of indirection no matter what (self.chunk_function -> split_sentence), what do you think about creating class method def chunk_function() that calls split_sentence (which remains its own function) instead? this still allows the user to overwrite the class method if they want to.

I'll push up my changes to illustrate what I mean.